app/controllers/github_integrations_controller.rb

```diff
@@ -21,20 +22,20 @@ def get_repositories
   def get_branches
     return fail_with_reason("User not connected to Github") unless @github_integration
     return fail_with_reason("Repository not provided") unless params["repository"]
+
     branches = @github_integration.branches(params["repository"])
     render json: branches, status: :ok
   rescue StandardError => e
     render json: { error: e.message }, status: :not_found
     nil
   end
 
-  private
+private
 
   def set_github_integration
-    gi = current_user.github_integration
-    if gi && gi.is_connected
-      @github_integration = current_user.github_integration
-    end
+    return if !current_user&.github_integration&.is_connected
```

Comment:
let me know if this is similar to what you intended.

Reply:
Could the if and negation be changed to an unless?

Reply:
yup that make sense